src/basic/chase-symlinks.c

```diff
@@ -552,14 +552,22 @@ int chase_symlinks_and_opendir(
                 return 0;
         }
 
-        r = chase_symlinks(path, root, chase_flags, ret_path ? &p : NULL, &path_fd);
+        r = chase_symlinks(path, root, chase_flags, &p, &path_fd);
         if (r < 0)
                 return r;
         assert(path_fd >= 0);
 
         d = opendir(FORMAT_PROC_FD_PATH(path_fd));
-        if (!d)
-                return -errno;
+        if (!d) {
+                /* Hmm, we have the fd already but we got ENOENT, most likely /proc is not mounted.
+                 * Let's try opendir() again on the full path. */
+                if (errno == ENOENT) {
+                        d = opendir(p);
+                        if (!d)
+                                return -errno;
+                } else
+                        return -errno;
```

Comment:
also, this is a very generic function, and this insecure fallback looks very wrong in mos cases. If anything this must be opt-in.